c++/src/kj/compat/http.c++

```diff
@@ -2663,10 +2674,11 @@ private:
     });
 
     // There's no buffered incoming data, so start pumping stream now.
-    return stream->pumpTo(*other.stream).then([&other](size_t) -> kj::Promise<void> {
+    return stream->pumpTo(*other.stream).then([this, &other](size_t s) -> kj::Promise<void> {
       // WebSocket pumps are expected to include end-of-stream.
       other.disconnected = true;
       other.stream->shutdownWrite();
+      sentBytes += s;
```

Comment:
Down here, you need to increment both `this->receivedBytes` as well as `other.sentBytes` to account for the bytes that were pumped directly from one underlying stream to the other.